azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/ConfigMojo.java

```diff
@@ -60,12 +59,22 @@
     public static final String JAVA_11 = "11";
     public static final String JAVA_11_STRING = "java 11";
 
+    private static final String PRICE_TIER_NOT_AVAIL = "The price tier \"P1\", \"P2\", \"P3\" are only available for Windows runtime, use \"%s\" instead.";
+
     private MavenPluginQueryer queryer;
     private WebAppPomHandler pomHandler;
     private static final String[] configTypes = {"Application", "Runtime", "DeploymentSlot"};
 
     @Override
     protected void doExecute() throws AzureExecutionException {
+        if (!(Utils.isPomPackagingProject(this.project.getPackaging()) ||
```

Comment:
could we create a new static filed store all the supported packaging and validate by `StringUtils::equalsAny` ?